### PR TITLE
Reserved Words

### PR DESCRIPTION
**Pull request by mheap**
Some languages contain reserved words which make it difficult to build integrations with APIs using those words as part of their terminology. 

This PR intends to start a list of words that cannot be used as key names in our API

### CODE REVIEW

File: content/standard/reserved-words.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+* `nil`
+* `not`
+* `or`
+* `redo`
+* `rescue`
+* `retry`
+* `return`
+* `self`
+* `super`
+* `then`
```
Comment (tozgun) [inline content/standard/reserved-words.md]:
```suggestion
* `switch`
* `then`
```

File: content/standard/reserved-words.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* `def`
+* `defined?`
+* `do`
+* `else`
+* `elsif`
+* `end`
+* `ensure`
+* `false`
+* `for`
+* `if`
```
Comment (tozgun) [inline content/standard/reserved-words.md]:
```suggestion
* `goto`
* `if`
```

File: content/standard/reserved-words.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+* `next`
+* `nil`
+* `not`
+* `or`
+* `redo`
+* `rescue`
+* `retry`
+* `return`
+* `self`
+* `super`
```
Comment (tozgun) [inline content/standard/reserved-words.md]:
```suggestion
* `string`
* `super`
```

**Comment by lornajane** [comment]
This looks great and I am +1 for a reserved words list. I'm a little worried we're ruling out a bunch of things that could be really useful as filter words (and I'm not sure why there would ever be confusion about a ruby command and user-supplied information :scream: ). I recommend that we remove from this list `start`, `end` and `retry`

**Comment by lornajane** [comment]
Just noticed that we also have `start` and `end` as recommended fieldnames https://nexmo.github.io/api-standards/standard/data-structure/ <-- adding this in case there are other words we want to check against the list

**Pull request closed without merging**